ampersand-calendar.js

```diff
@@ -0,0 +1,304 @@
+;(function() {
+  var d3 = require('d3');
+  var _ = require('lodash');
+  var moment = require('moment');
+
+  var AmpersandState = require('ampersand-state');
+  var AmpersandView = require('ampersand-view');
+
+  var CalendarState = AmpersandState.extend({
+    props: {
+      startDate: [ 'object', false, function() { return null; } ],
+      endDate: [ 'object', false, function() { return null; } ]
+    },
+    session: {
+      current: [ 'object', false, function() { return moment(); } ],
+      currentDates: 'array'
```

Comment:
From what I can tell, this should be a derived property that depends on `current`, `startDate`, and `endDate` and just returns `this.createDateArray()`. That would keep you from having to assign it all over the place.